queue-load-leveling/src/main/java/org/queue/load/leveling/TaskGenerator.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * The MIT License
+ * Copyright (c) 2014 Ilkka Seppälä
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.queue.load.leveling;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * TaskGenerator class.
+ * Each TaskGenerator thread will be a Worker which submit's messages to the queue.
+ * We need to mention the message count for each of the TaskGenerator threads.
+ * 
+*/
+public class TaskGenerator implements Task, Runnable {
+
+  
+  private static final Logger LOGGER = LoggerFactory.getLogger(App.class);
+  
+  // MessageQueue reference using which we will submit our messages.
+  private MessageQueue msgQueue;
+  
+  // Total message count that a TaskGenerator will submit.
+  private int msgCount;
```

Comment:
I am using `msgCount` to create particular number of messages which I am submitting to MessageQueue. For each message I am reducing the the `msgCount` value. If I make this value final then I can't keep count of message that I am creating. How to solve this issue?